09_maps/main.go

```diff
@@ -111,18 +122,129 @@ func main() {
 	// Print the BMI of each person in the array.
 	// Hint: Use a loop.
 
+	// create empty array of maps
+	var people []map[string]int = []map[string]int{}
+	// fill array of maps
+	// people[0]["height"] = 180
+	// people[0]["weight"] = 70
+	people = append(people, map[string]int{"height": 180, "weight": 70})
+	people = append(people, map[string]int{"height": 160, "weight": 50})
+	people = append(people, map[string]int{"height": 170, "weight": 80})
+	people = append(people, map[string]int{"height": 150, "weight": 60})
+	people = append(people, map[string]int{"height": 190, "weight": 90})
+	// print people[0]
+	fmt.Println(people)
+	// print BMI
+	calcBMI(people[0])
+	// var counter int = 0
+	// // var calcPeople []map[string]int = people[counter]
+
+	// for {
+	// 	if counter >= len(people) {
+	// 		break
+	// 	}
+	// 	// print bmi by calling the calcBMI with an array value of people
+	// 	// calcBMI(calcPeople)
+	// 	calcBMI(people[counter])
+	// 	counter++
+	// }
+
 	// For later exercises, use the same format as the previous one.
 
-	// TODO: Create a function that takes an array of maps as an argument and
-	// prints the average BMI of the people in the array.
+	// Close main function, to be able to declare another function
+}
 
-	// TODO: Create a function that takes an array of maps as an argument and
-	// returns the map with the highest BMI.
+// create empty map
+// create function
+// create counter
+// create loop
+// create break condition if counter is > than the key-value pairs the map has
+// operate on the input value of the keys and do math on it
+// add the result to a new map
+// add a new key value element and fill it depending on the value of the input height key
+// print new map
+
+var newMap map[string]int = map[string]int{}
+
+func createNewMap(mapInput map[string]int) {
+	var counter int = 0
+
+	for {
+		if counter > len(mapInput) {
+			break
+		}
+		newMap["age"] = mapInput["age"] + 2
+		newMap["height"] = mapInput["height"] * 2
+		newMap["weight"] = mapInput["weight"] * 2
+		if mapInput["height"] > 200 {
+			newMap["tall"] = 1
+		}
+		newMap["tall"] = 0
+		counter++
+	}
+	fmt.Println("createNewMap:", newMap)
+}
 
-	// TODO: Create a function that takes an array of maps as an argument and
-	// add as a new key-value pair to each map the key "bmi" with the value
-	// being the BMI of the corresponding person.
-	// Hint: Use a loop.
-	// Hint: Create a new map for each person.
-	// Hint: Create a new array of maps.
+// TODO: Create a function that takes a map as an argument and prints the
+// BMI of the person. The formula for BMI is:
+// BMI = weight / (height * height)
+// If the map does not have the key "height", print "Height not found".
+// If the map does not have the key "weight", print "Weight not found".
+// Otherwise, print "<name>'s BMI is <bmi>".
+// Hint: Use this function in later exercises.
+
+func calcBMI(mapInput map[string]int) {
+	var value int
+	var exists bool
+	var bmi float32
```

Comment:
Why not `float64`?

Reply:
Didn't read too much into the difference of `float32` vs `float64`.
Changed.